pkg/edhoc-c: initial commit (#16295)
RIOT-OS/RIOT: pkg/edhoc-c/doc.txt

/**
 * @defgroup pkg_edhoc-c EDHOC-C
 * @ingroup  pkg
 * @brief    Support for Ephemeral Diffie-Hellman Over COSE (EDHOC)
 * @see      https://github.com/openwsn-berkeley/EDHOC-C
 *
 * # EDHOC-C RIOT package
 *
 * ## Usage
 *
 * Just add it as a package in your application:
 *
 * ```makefile
 * USEPKG += edhoc-c
 * ```
 *
 * ### Backends
 *
 * EDHOC-C does not implement crypographic algorithms, instead it uses external
 * libraries as a backend. Libaries that provide all requirements are
 * currently only @ref pkg_wolfssl, its selected through the following module:
 *
 * ```makefile
 * USEMODULE += edhoc-c_crypto_wolfssl
 * ```
 *
 * EDHOC-C does not implement a CBOR library either, possible backends are
 * @ref pkg_nanocbor and @ref pkg_cn-cbor. Pick one of them through the
 * following modules:
 *
 * ```makefile
 * USEMODULE += edhoc-c_cbor_nanocbor
 * USEMODULE += edhoc-c_cbor_cn_cbor
 * ```
 *
 * And don't forget to include the header for the EDHOC-C public API:
 *
 * ```c
 * #include <edhoc/edhoc.h>
 * ```
 *
 * ### Current Support
 *

>  * - EDHOC-C support @ref pkg_hacl as a crypto backend but its using a different

Reviewer: ```suggestion
 * - EDHOC-C supports @ref pkg_hacl as crypto backend but it's using a different
```

 *   version than the one supported currently in RIOT

>  * - x509 certificates required MBED-TLS, which is currently not supported in

Reviewer: ```suggestion
 * - x509 certificates require MBED-TLS, which is currently not supported in
```

>  *   RIOT so only RPK and CBOR certificates are possible.

Reviewer: ```suggestion
 *   RIOT, so only RPK and CBOR certificates are supported.
```
